src/etc/frontend/di.xml

```diff
@@ -1,5 +1,14 @@
+<?xml version="1.0"?>
 <config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:ObjectManager/etc/config.xsd">
     <type name="Magento\Checkout\Controller\Index\Index">
         <plugin name="checkout_checkout_index_action_plugin" type="Hyva\Checkout\Plugin\CheckoutController\Index"/>
     </type>
+    <type name="Magento\Customer\Controller\Ajax\Login">
+        <arguments>
+            <argument name="resultJsonFactory"
+                      xsi:type="object">Hyva\Checkout\Result\JsonFactory</argument>
```

Comment:
I thank Alan for his great post about dependency injection which I read a couple of years back. Otherwise, who will know there is an argument replace feature available in dependency injection 🤣